src/Uno.Toolkit.RuntimeTests/Tests/DrawerTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.UI.RuntimeTests;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.Extensions;
+using Windows.ApplicationModel.Appointments;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests;
+
+[TestClass]
+[RunsOnUIThread]
+internal class DrawerTests
+{
+	[TestMethod]
+#if __IOS__
+	[Ignore("Opacity is not supposed to be modified from non ui-thread")]
+#endif
+	public async Task IsOpen_FromNonUIThread()
+	{
+		var drawer = new DrawerControl
+		{
+			Content = new Grid()
+		};
+
+		// don't wait for loaded, start the task immediately
+		UIHelper.Content = drawer;
+		await Task.Run(async () =>
+		{
+			drawer.IsOpen = true;
+			await Task.Delay(2000);
+			drawer.IsOpen = false;
+		});
+
+		// leave time for IsOpen=false (animation or not) to finish (if it doesn't throw)
+		await Task.Delay(1000);
```

Comment:
Please avoid `Task.Delay` in tests ... this cause CI to take hours. Are we able to have a template that has a duration of 0? Or just a minimal delay (to be sure to be async, something like 10ms) with pooling on the the assertion with a timeout (cf. `UIHelper.WaitFor` in uno)?

Reply:
Often just yielding few times is enough for the queue to process all messages and it won't cause a multi-hours CI time.